Add Kong Gateway Operator docs (#6144)
Kong/docs.konghq.com: app/_src/gateway-operator/customization/pod-template-spec.md

---
title: Using PodTemplateSpec
---

> The `DataPlane` resource uses the Kubernetes [PodTemplateSpec](/gateway-operator/{{ page.release }}/customization/pod-template-spec/) to define how pods should be run. This allows you to customize everything you need to in your deployment.

Reviewer: ```suggestion
The `DataPlane` resource uses the Kubernetes [PodTemplateSpec](/gateway-operator/{{ page.release }}/customization/pod-template-spec/) to define how pods should run. You can customize everything you need to in your deployment using this resource.
```

> For detailed documentation see the [PodTemplateSpec documentation](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-template-v1/#PodTemplateSpec) on the Kubernetes docs.

Reviewer: ```suggestion
For more information about creating copies of a predefined Pod, see the [PodTemplateSpec documentation](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-template-v1/#PodTemplateSpec).
```

> We have dedicated examples for the most common applications of `PodTemplateSpec`:

Reviewer: ```suggestion
Here are some examples for the most common use of `PodTemplateSpec`:
```

* [Customize the DataPlane image](/gateway-operator/{{ page.release }}/customization/data-plane-image/)
* [Deploy a sidecar container](/gateway-operator/{{ page.release }}/customization/sidecars/)

> You might also want to customize the following:

Reviewer: ```suggestion
In addition, you can also customize:
```

* Volume mounts
* Node affinity / anti-affinity

> * Service account used to run the pod

Reviewer: ```suggestion
* Service account used to run the Pod
```

> If you need help customizing your deployment, please raise an issue in [Kong/gateway-operator-docs/](https://github.com/Kong/gateway-operator-docs/issues).

Reviewer: ```suggestion
If you need additional information about customizing your deployment, raise an issue in [Kong/gateway-operator-docs/](https://github.com/Kong/gateway-operator-docs/issues).
```
